components/resources/quizzes/careerQuiz/career-quiz.tsx

```diff
@@ -83,14 +93,50 @@ const CareerQuiz = () => {
       } else setStage((prevStage) => prevStage - 1);
       setTriggerAnimation(true);
     }, 250); // adjust the delay time based on the animation duration
+    window.scrollTo({
+      top: 0,
+      left: 0,
+      behavior: "smooth",
+    });
   };
 
   const handleOptionClick = (option: QuizOptionViewState) => {
+    const currentQuestion =
+      quizViewState.questions[quizViewState.currentQuestion];
+    const selectedOptions = currentQuestion.options.filter(
+      (option) => option.isSelected
+    );
+    const maxSelection =
+      quizData.questions[quizViewState.currentQuestion].maxSelections;
+    // MaxSelection ===1 edgecase
+    if (maxSelection === 1) {
+      // Set all other options to false and set the selected option to true
+      const selectedQuizOption = quizViewState.questions.map((question) => ({
+        ...question,
+        options: question.options.map((questionOption) =>
+          questionOption === option
+            ? { ...questionOption, isSelected: true }
+            : { ...questionOption, isSelected: false }
+        ),
+      }));
+
+      const updatedQuizViewState = {
+        ...quizViewState,
+        questions: selectedQuizOption,
+      };
+      setQuizViewState(updatedQuizViewState);
+      return;
+    } else if (selectedOptions.length >= maxSelection && !option.isSelected) {
+      return;
+    }
     const selectedQuizOption = quizViewState.questions.map((question) => ({
```

Comment:
I might wrap the rest of this code in an else statement, if that makes sense to you.